packages/dev/core/src/Meshes/greasedLineMesh.ts

```diff
@@ -0,0 +1,535 @@
+        const cloned = new GreasedLineMesh(name, this._scene, <GreasedLineMeshOptions>lineOptions, pluginMaterial);
+        cloned.material = material;
+
+        if (newParent) {
+            cloned.parent = newParent;
+        }
+
+        return cloned;
+    }
+
+    /**
+     * Serializes this ground mesh
+     * @param serializationObject object to write serialization to
+     */
+    public serialize(serializationObject: any): void {
+        super.serialize(serializationObject);
+        serializationObject.lineOptions = this._options;
+        serializationObject.materialOptions = this._pluginMaterial.getOptions();
+    }
+
+    /**
+     * Parses a serialized ground mesh
+     * @param serializedMesh the serialized mesh
+     * @param scene the scene to create the ground mesh in
+     * @returns the created ground mesh
+     */
+    public static Parse(serializedMesh: any, scene: Scene): Mesh {
+        const lineOptions = <GreasedLineMeshOptions>serializedMesh.lineOptions;
+        const materialOptions = <GreasedLineMaterialOptions>serializedMesh.materialOptions;
+        const name = <string>serializedMesh.name;
+
+        const material =
+            materialOptions.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR ? new PBRMaterial(name, scene) : new StandardMaterial(name, scene);
+        const pluginMaterial = new GreasedLinePluginMaterial(material, scene, materialOptions);
+
+        const parsed = new GreasedLineMesh(name, scene, lineOptions, pluginMaterial);
+        parsed.material = material;
+        return parsed;
+    }
+
+    // TODO: which parameters to suppport?
+    public intersects(
+        ray: Ray,
+        fastCheck?: boolean | undefined,
+        trianglePredicate?: TrianglePickingPredicate | undefined,
+        onlyBoundingInfo?: boolean,
+        worldToUse?: Matrix | undefined,
+        skipBoundingInfo?: boolean
+    ): PickingInfo {
+        const pickingInfo = new PickingInfo();
+        const intersections = this.intersections(ray);
+        if (intersections?.length === 1) {
+            const intersection = intersections[0];
+            pickingInfo.hit = true;
+            pickingInfo.distance = intersection.distance;
+            pickingInfo.ray = ray;
+            pickingInfo.pickedMesh = intersection.object;
+            pickingInfo.pickedPoint = intersection.point;
+        }
+
+        return pickingInfo;
+    }
+
+    /**
+     * Gets all intersections of a ray and the line.
+     * @param ray Ray
+     * @param firstOnly If true, the first and only intersection is immediatelly returned if found.
+     * @returns intersection(s)
+     */
+    public intersections(ray: Ray, firstOnly = true): { distance: number; point: Vector3; object: GreasedLineMesh }[] | undefined {
+        if (this._boundingSphere && ray.intersectsSphere(this._boundingSphere, this.intersectionThreshold) === false) {
+            return;
+        }
+
+        const vStart = new Vector3();

```

Comment:
Try as much as possible to limit the creation of new objects to limit the GC, because this function could be called several times per frame (use `TmpVectors.Vector3[..]` instead).

Reply:
Didn't know about it. I will definitelly switch to the recommended function.